docs/pages/choose-an-edition/migrate-to-cloud.mdx

```diff
@@ -0,0 +1,341 @@
+resources should be considered for migration:
+
+ - Teleport agents 
+ - Machine ID Bots
+ - Access Request plugins
+ - The Teleport Event Handler
+
+Before migrating services, make sure you are logged in to your new cloud-hosted
+Teleport Enterprise account.
+
+You can migrate Teleport services all at once or gradually, depending on your
+business requirements. If running Teleport at scale, you'll generally want to
+use a configuration management tool to automate and streamline the process of
+carrying out the actions involved in migrating agent configurations. 
+
+### Migrate Teleport agents
+
+To migrate Teleport agents:
+
+1. For each agent and Machine ID bot, obtain a valid join token. We recommend
+   using a [delegated join
+   method](https://goteleport.com/docs/reference/join-methods/#delegated-join-methods).
+
+1. If using ephemeral tokens, ensure you specify the appropriate token type to
+   match the Teleport services. Token types can include `node`, `app`, `kube`,
+   `db`, `windowsdesktop` and others depending on the service you wish to join
+   to your Teleport Cluster.
+
+1. In the following example, a new token is created with the TTL of fifteen
+   minutes:
+
+   ```code
+   $ tctl tokens add --type node,app,db --ttl 15m
+   ```
+
+   In this command, we assigned the token the `node`, `app` and `db` type,
+   indicating that it will allow an agent to join which is running the Teleport
+   `ssh_service`, `db_service` and `app_service`.
+
+   Copy the token so you can use it later in this guide.
+
+1. Stop Teleport services on the agent (if applicable).
+
+1. Update the `proxy_server` or `auth_servers` field in the agent configuration
+   file to point to the address of your cloud-hosted Teleport Enterprise
+   cluster. By default, on Linux servers, the configuration is located in the
+   `/etc/teleport.yaml` directory:
+
+   ```yaml
+   version: v3
+   teleport:
+      proxy_server: example.teleport.sh:443
+   ```
+
+   If your agent configuration does _not_ include a `teleport.proxy_server`
+   field, and instead has a `teleport.auth_server` or `teleport.auth_servers`
+   field, we recommend migrating your configuration to `version: v3` and using
+   `teleport.proxy_server`. 
+
+   With the `teleport.proxy_server` field, the agent attempts to connect to the
+   Teleport cluster using a single mode, rather than multiple modes, which takes
+   less time and involves less functionality to troubleshoot.
+
+1. Update the `auth_token` or `join_params.token_name` field with the newly
+   generated token.
+
+   ```yaml
+   teleport:
+     join_params:
+       method: token
+       token_name: new-token-goes-here
+   ```
+
+1. On Linux servers, delete the local agent cache to force the agent to rejoin
+   the new cloud-hosted Teleport Enterprise cluster. By default, the data is
+   located in the  `/var/lib/teleport` directory:
+
+   ```code
+   rm -rf /var/lib/teleport
+   ```

```

Comment:
We're also missing instructions here about how to wipe out existing state to rejoin a Kubernetes agent to a new cluster. Essentially it's something like:

```
# get the release name
helm -n <namespace> ls
# delete the state secret
kubectl -n <namespace> delete secret <release-name>-0-state
```